astropy/table/table.py

```diff
@@ -785,35 +785,55 @@ def _new_from_slice(self, slice_):
         table.meta.clear()
         table.meta.update(deepcopy(self.meta))
         table.primary_key = self.primary_key
-        cols = self.columns.values()
 
         newcols = []
-        for col in cols:
-            col.info._copy_indices = self._copy_indices
+        for col in self.columns.values():
             newcol = col[slice_]
-            if col.info.indices:
+
+            # Note in line below, use direct attribute access to col.indices for Column
+            # instances instead of the generic col.info.indices.  This saves about 4 usec
+            # per column.
```

Comment:
Yikes. Maybe add "TODO: make the info implementation much faster." For instance, where in `info` we know that an attribute is from the parent, we can explicitly create a property that accesses it rather than rely on `__getattr__`.  Anyway, for another PR!

Reply:
Would be nice if there were a way to tell whether any indices are set at all on the table itself...